### PR TITLE
Overmap respects circular distances

### PR DESCRIPTION
**Pull request by thquinn**
#### Summary
```SUMMARY: Bugfixes "Overmap respects circular distances"```

#### Purpose of change
Overmap revealing is currently square when circular distances are enabled.

#### Describe the solution
When circular distances are enabled, I increase the see cost of overmap tiles based on how diagonal the line of sight is by calculating the length of the line segment with that angle intersecting the center of a unit square. I also changed ```overmapbuffer::reveal()``` to reveal a circular area when circular distances are enabled.

ss

ss2

#### Describe alternatives you've considered
For a more accurate measurement of see cost, we could iterate over each tile crossed by the line of sight, not just those in the Bresenham line. We would then calculate the length of the segment that intersects with that tile and use that as the see-cost multiplier. I figured this was a little overcomplicated, and since all other lines in the game are discretized to tiles, there wasn't much reason to get closer to a continuous line here.

### CODE REVIEW

**Comment by ZhilkinSerg** [comment]
Can you make this respect `CIRCLEDIST` option?

https://github.com/CleverRaven/Cataclysm-DDA/blob/06a738afbb7a7402023101ecce34c7f83dedf4e1/src/options.cpp#L1153-L1156

**Comment by thquinn** [comment]
Updated description to include that I'm only doing this when the Circular Distances option is enabled.

File: src/game.cpp [hunk 1]
```diff
@@ -11762,16 +11762,33 @@ void game::update_overmap_seen()
...
+        for( int dy = -dist; dy <= dist; dy++ ) {
+            if( trigdist && dx * dx + dy * dy > dist_squared ) {
+                continue;
+            }
+            int x = ompos.x + dx;
+            int y = ompos.y + dy;
+            float multiplier;
+            if( trigdist ) {
+                float angle = atan2( y - ompos.y, x - ompos.x ) + M_PI;
+                angle = fmod( angle, M_PI / 2 );
```
Comment (Night-Pryanik) [inline src/game.cpp]:
```suggestion
               float angle = fmod( atan2( y - ompos.y, x - ompos.x ) + M_PI, M_PI / 2 );
```
And remove the upper line with declaration of `angle`.

File: src/game.cpp [hunk 2]
```diff
@@ -11762,16 +11762,32 @@ void game::update_overmap_seen()
...
+            }
+            int x = ompos.x + dx;
+            int y = ompos.y + dy;
+            float multiplier;
+            if( trigdist ) {
+                float angle = fmod( atan2( dy, dx ) + M_PI, M_PI / 2 );
+                if( angle > M_PI / 4 ) {
+                    angle = M_PI / 2 - angle;
+                }
+                multiplier = 1 / cos( angle );
```
Comment (codemime) [inline src/game.cpp]: You probably don't need any of these trigonometric functions and angle adjustments. The math here is fairly simple:
```c++

const int h_squared = dx * dx + dy * dy;

if( h_squared == 0 || h_squared > dist_squared )
    continue;
    
const float multiplier = std::sqrt( h_squared ) / std::max<float>( std::abs( dx ), std::abs( dy ) );
```

Comment (thquinn) [reply src/game.cpp]: I guess it is just that simple, huh... I'm going to do another overmap vision PR soon; I'm happy to sneak this refactor in with that.

**Pull request merged**